MetaMask/metamask-extension
# feat: poll native currency prices across chains

## **Description**

Leverages https://github.com/MetaMask/core/pull/4852 to fetch native currency prices across all evm chains, instead of just the current chain.

Metamask will now hit the `/pricemulti` endpoint of cryptocompare to fetch an array of currencies in 1 request.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/28196?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

No user facing changes.

1. Add some networks with native currencies other than ETH like polygon, bnb
2. Verify native tokens have the correct fiat price
3. Export state and verify `currencyRates` has entries for each native currency

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
